astropy/io/fits/file.py

```diff
@@ -491,7 +484,7 @@ def _open_filename(self, filename, mode, overwrite):
             with fileobj_open(self.name, 'rb') as f:
                 magic = f.read(4)
         else:
-            magic = b('')
+            magic = ''.encode("latin-1")
```

Comment:
Why 'latin-1'?

Reply:
Should be simply `b''` *i.e.* an empty byte string.

Reply:
Oh, sure. It was one of those blind updates based on this:
``A “fake” bytes literal. data should always be a normal string literal. In Python 2, b() returns a 8-bit string. In Python 3, data is encoded with the latin-1 encoding to bytes.``